// @redq/reuse-modal/lib/hooks/useRnd.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.default = useRnd;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRnd = require('react-rnd');

var _useClickOutside = require('./useClickOutside');

var _useClickOutside2 = _interopRequireDefault(_useClickOutside);

var _useWindowSize = require('./useWindowSize');

var _useWindowSize2 = _interopRequireDefault(_useWindowSize);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function useRnd() {
  var props = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var dispatch = props.dispatch,
      closeOnClickOutside = props.closeOnClickOutside;


  var size = process.browser && (0, _useWindowSize2.default)();
  return function RND(props) {
    var modalRef = (0, _react.useRef)();
    (0, _useClickOutside2.default)({ modalRef: modalRef, dispatch: dispatch, closeOnClickOutside: closeOnClickOutside });
    var config = props.config;

    var rndConfig = _extends({}, config, {
      className: 'reuseModalHolder ' + config.className,
      default: {
        width: config.default.width,
        height: config.default.height,
        x: config.default.x || (size.width - config.default.width) / 2,
        y: config.default.y || (size.height - config.default.height) / 2
      }
    });
    return _react2.default.createElement(
      _reactRnd.Rnd,
      _extends({}, rndConfig, { ref: modalRef }),
      _react2.default.createElement(
        'div',
        { className: 'innerRndComponent' },
        props.children
      )
    );
  };
}
module.exports = exports['default'];
